client/verta/verta/endpoint/_endpoint.py

```diff
@@ -509,8 +509,15 @@ def _set_kafka_settings(self, stage_id, kafka_settings):
                 " not {}".format(type(kafka_settings))
             )
         if kafka_settings._cluster_config_id is None:
-            config = kafka.list_kafka_configurations(self._conn)[0]
-            kafka_settings._cluster_config_id = config.get("id", None) if config else None
+            configs = kafka.list_kafka_configurations(self._conn)
+            if not configs:
+                raise RuntimeError("no valid Kafka configuration found.")
+            config_id = configs[0].get("id", None)
+            if config_id is None:
+                raise RuntimeError(
+                    "Kafka configuration is missing the required cluster_configuration_id."
```

Comment:
This error message kind of sounds like the client user neglected to provide something, but it's actually bad state in our backend that they can't do anything about. Also not sure where "cluster_configuration_id" came from, because I don't think we have that field name anywhere 🤔 

Maybe

```suggestion
                    "active Kafka configuration is missing its ID; please notify the Verta development team"
```